mitosheet/mitosheet/public/v3/sheet_functions/bool_functions.py

```diff
@@ -123,6 +123,52 @@ def IF(condition: pd.Series, true_series: AnyPrimitiveOrSeriesInputType, false_s
     )
 
 
+@cast_values_in_arg_to_type('argv', 'bool')
+@handle_sheet_function_errors
+def IFS(*argv: Optional[IfsInputType]) -> pd.Series:
+    """
+    {
+        "function": "IFS",
+        "description": "Returns one value if the condition is True. Returns the other value if the conditon is False.",
+        "search_terms": ["ifs", "if", "conditional", "and", "or"],
+        "examples": [
+            "IFS(Status == 'success', 1, 0)",
+            "IFS(Nums > 100, 100, Nums)",
+            "IFS(AND(Grade >= .6, Status == 'active'), 'pass', 'fail')"
+        ],
+        "syntax": "IFS(boolean_condition_1, value_if_true, boolean_condition_2, value_if_true)",
+        "syntax_elements": [{
+                "element": "boolean_condition",
+                "description": "An expression or series that returns True or False values. Valid conditions for comparison include ==, !=, >, <, >=, <=."
+            },
+            {
+                "element": "value_if_true",
+                "description": "The value the function returns if condition is True."
+            }
+        ]
+    }
+    """
+    # Go through all of the conditions. The first condition that is true for
+    # *each cell* should be selected. 
+    results = pd.Series()
+    for index, condition in enumerate(argv):
+        # The args are alternating between the conditions and the values corresponding to them.
```

Comment:
I think that we can actually do input sanity checking. Namely, all series with an even index should be boolean (or a boolean series).

Reply:
You can see an example of how a sheet function might raise an error in the `SUMPRODUCT` function in the v3 interface.

Reply:
Just added it (and a test for the error)! How does that look?